modules/restore-replace-stopped-etcd-member.adoc

```diff
@@ -0,0 +1,147 @@
++
+If you are running an installer-provision installation, or you used the Machine API to create your machines, follow these steps. Otherwise, you must create the new master using the same method that was used to originally create it.
+
+.. Obtain the machine for the unhealthy member.
++
+In a terminal that has access to the cluster as a `cluster-admin` user, run the following command:
++
+----
+$ oc get machines -n openshift-machine-api -o wide
+
+NAME                                        PHASE     TYPE        REGION      ZONE         AGE     NODE                           PROVIDERID                              STATE
+clustername-8qw5l-master-0                  Running   m4.xlarge   us-east-1   us-east-1a   3h37m   ip-10-0-131-183.ec2.internal   aws:///us-east-1a/i-0ec2782f8287dfb7e   stopped <1>
+clustername-8qw5l-master-1                  Running   m4.xlarge   us-east-1   us-east-1b   3h37m   ip-10-0-154-204.ec2.internal   aws:///us-east-1b/i-096c349b700a19631   running
+clustername-8qw5l-master-2                  Running   m4.xlarge   us-east-1   us-east-1c   3h37m   ip-10-0-164-97.ec2.internal    aws:///us-east-1c/i-02626f1dba9ed5bba   running
+clustername-8qw5l-worker-us-east-1a-wbtgd   Running   m4.large    us-east-1   us-east-1a   3h28m   ip-10-0-129-226.ec2.internal   aws:///us-east-1a/i-010ef6279b4662ced   running
+clustername-8qw5l-worker-us-east-1b-lrdxb   Running   m4.large    us-east-1   us-east-1b   3h28m   ip-10-0-144-248.ec2.internal   aws:///us-east-1b/i-0cb45ac45a166173b   running
+clustername-8qw5l-worker-us-east-1c-pkg26   Running   m4.large    us-east-1   us-east-1c   3h28m   ip-10-0-170-181.ec2.internal   aws:///us-east-1c/i-06861c00007751b0a   running
+----
+<1> This is the master machine for the unhealthy node, `ip-10-0-131-183.ec2.internal`.
+
+.. Save the machine configuration to a file on your file system:
++
+----
+$ oc get machine clustername-8qw5l-master-0 -n openshift-machine-api -o yaml > new-master-machine.yaml
+----
++
+// TODO: Put on separate lines to call out machine name?
+
+.. Edit the `new-master-machine.yaml` file that was created in the previous step.
+
+... Remove the entire `status` section.
+
+... Remove the `providerID` field.
+
+... Change the `name` field to a new name.
++
+It is recommended to keep the same base name as the old machine and change the ending number to the next available number. In this example, `clustername-8qw5l-master-0` is changed to `clustername-8qw5l-master-3`.
+
+... Update the `selfLink` field to use the new machine name from the previous step.
+
+.. Delete the machine of the unhealthy member:
++
+----
+$ oc delete machine clustername-8qw5l-master-0 -n openshift-machine-api
+----
+
+.. Wait until the machine is deleted before proceeding to the next step:
++
+----
+$ until $(oc get machine clustername-8qw5l-master-0 -n openshift-machine-api | grep "No resources found"); do sleep 1; echo "Waiting for the machine to be deleted"; done
+----
```

Comment:
@hexfusion so this will watch forever, even once the machine has been deleted?

Could we just run that command w/o watching, (`oc get machines -n openshift-machine-api`) and have them proceed with adding the new machine once the old machine is gone from the list?

Reply:
absolutely fine with me, the point is simple commands.